### PR TITLE
4796 add facility to vaccination

### LINKED ISSUE #4796: Add facility to vaccination
## Is your feature request related to a problem? Please describe 👀

I want to know where the vaccination I am recording has occurred. For now, default to this facility, or if desired, switch a free text field. In future, users will be able to select other facilities as well:

## Describe the solution you'd like 🎁

Add facility field to vaccination modal: 

Screenshot 2024-09-16 at 11 48 28 AM

### PR DESCRIPTION
**Pull request by lache-melvin**
Fixes #4796

# 👩🏻‍💻 What does this PR do?

Adds inputs to the vaccination modal for this facility/free text other facility.

Screenshot 2024-09-20 at 6 00 32 PM
Screenshot 2024-09-20 at 6 00 41 PM

## 💌 Any notes for the reviewer?

Haven't done the part of not requiring item/batch when its for another facility. Will do that with #4801, as need to make some changes to the "batch is required" logic...

# 🧪 Testing

- [ ] Add a vaccination for your facility
- [ ] Add a vaccination for a different facility - you can input custom name
- [ ] Edit a vaccination, you can change to/from "other"

# 📃 Documentation

- [x] **Part of an epic**: documentation will be completed for the feature as a whole
- [ ] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [ ] **These areas should be updated or checked**: 
  1.
  2.

### CODE REVIEW

File: server/graphql/programs/src/mutations/vaccination/insert.rs [hunk 1]
```diff
@@ -94,6 +100,7 @@ fn map_error(error: ServiceError) -> Result<InsertVaccinationResponse> {
     let graphql_error = match error {
         ServiceError::VaccinationAlreadyExists
         | ServiceError::ClinicianDoesNotExist
+        | ServiceError::FacilityNameDoesNotExist
```
Comment (CarlosNZ) [inline server/graphql/programs/src/mutations/vaccination/insert.rs:103]: Small nit, but could we call this "FacilityDoesNotExist"? It makes it seem like we're checking the name field, but we're not, we're checking that the ID exists as a Facility name (right?) (I was initially confused as I wondered why we'd check the name for a free text field). (Plus it's slightly more consistent with the "ClinicianDoesNotExist" right above it)

Comment (CarlosNZ) [inline server/graphql/programs/src/mutations/vaccination/insert.rs:103]: Small nit, but could we call this "FacilityDoesNotExist"? It makes it seem like we're checking the name field, but we're not, we're checking that the ID exists as a Facility name (right?) (I was initially confused as I wondered why we'd check the name for a free text field). (Plus it's slightly more consistent with the "ClinicianDoesNotExist" right above it)

**Review by CarlosNZ — APPROVED** [review]
Nice. :)

Couple of very small nit picks, update as you see fit and merge away :)

File: client/packages/system/src/Vaccination/api/useVaccinationsGraphQL.ts [hunk 2]
```diff
@@ -8,8 +8,8 @@ import { getSdk } from './operations.generated';
 export const useVaccinationsGraphQL = () => {
   const { client } = useGql();
   const queryClient = useQueryClient();
-  const { storeId } = useAuthContext();
+  const { storeId, store } = useAuthContext();
   const api = getSdk(client);
 
-  return { api, queryClient, storeId };
+  return { api, queryClient, storeId, store };
```
[the remark this answers is not in the document]
Comment (lache-melvin) [reply client/packages/system/src/Vaccination/api/useVaccinationsGraphQL.ts:14]: I tried... storeId is required in the graphql call, but the whole `store` node is possibly undefined. Was a bit icky having the is defined check everywhere, so just pulled both off 🤷‍♀️

**Pull request merged**